sky/backends/backend_utils.py

```diff
@@ -838,6 +838,7 @@ def write_cluster_config(
                 'cluster_name': cluster_name,
                 'num_nodes': num_nodes,
                 'disk_size': to_provision.disk_size,
+                'disk_type': to_provision.get_cloud_disk_type(),
```

Comment:
Correct me if I am wrong, we don't have to prompt the return value up to here. Instead, having the disk type in the `resources_vars` in L836 should be enough, as the upper level logics may not need to know the cloud specific disk_type specs.

Reply:
One principle for the class design is to reduce the number of interfaces. : )